src/scheduler/workers/main.ts

```diff
@@ -0,0 +1,264 @@
+    let isEmailSent = false;
+    const todayDateString = SCHEDULED_PROCESS_TODAY_DATE;
+    // todayDateString = formatDate(subDays(new Date(), 28)); // today is annual review start date
+    // todayDateString = formatDate(subDays(new Date(), 27)); // today is day before annual review start date
+
+    // email the posts and providers if today = one of the annual review milestone date
+    switch (todayDateString) {
+      case annualReviewKeyDates?.POST_ONE_MONTH:
+        isEmailSent = isEmailSentBefore(audit, "sendOneMonthPostEmail");
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "POST_ONE_MONTH", "sendOneMonthPostEmail");
+        }
+        break;
+      case annualReviewKeyDates?.POST_ONE_WEEK:
+        isEmailSent = isEmailSentBefore(audit, "sendOneWeekPostEmail");
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "POST_ONE_WEEK", "sendOneWeekPostEmail");
+        }
+        break;
+      case annualReviewKeyDates?.POST_ONE_DAY:
+        isEmailSent = isEmailSentBefore(audit as Audit, "sendOneDayPostEmail");
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "POST_ONE_DAY", "sendOneDayPostEmail");
+        }
+        break;
+      case annualReviewKeyDates?.START:
+        // email posts to notify of annual review start
+        isEmailSent = isEmailSentBefore(audit as Audit, "sendStartedPostEmail");
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "START", "sendStartedPostEmail");
+        }
+
+        // update ListItem.isAnnualReview if today = the START milestone date
+        // email providers to notify of annual review start
+        isEmailSent = isEmailSentBefore(listItemAudit as Audit, "sendStartedProviderEmail");
+        if (!isEmailSent) {
+          const updatedListItems = await updateIsAnnualReviewForListItems(listItemsForList, list);
+          const currentDateString = formatDate(addDays(parseDate(todayDateString), 42));
+          await processProviderEmailsForListItems(list, updatedListItems, "START", currentDateString);
+        }
+        break;
+    }
+  }
+}
+
+export async function updateIsAnnualReviewForListItems(listItems: ListItemWithHistory[], list: List): Promise<ListItemWithHistory[]> {
+  if (listItems.length === 0) {
+    logger.info(`No List items found for list ${list.id}`);
+    return [];
+  }
+  const updatedListItems: Result<ListItemWithHistory[]> = await updateIsAnnualReview(
+    list,
+    listItems,
+    ListItemEvent.ANNUAL_REVIEW_STARTED,
+    "startAnnualReview",
+    AuditEvent.REMINDER
+  );
+
+  if (updatedListItems.error) {
+    logger.info(`Unable to update list items and email providers for annual review start`);
+    return [];
+  }
+  if (!updatedListItems.result?.length) {
+    logger.info(`No list items were updated for annual review start`);
+    return [];
+  }
+  if (updatedListItems.result.length < listItems.length) {
```

Comment:
is this expected? what would the reason be that updatedListItems < listItems? since it's running in a transaction shouldn't it fail (and therefore be caught by updatedListItems.error)?

Reply:
the transaction is in a loop and is scoped for a single listItem and listItemEvent.  There could be a discrepancy where updatedListItems < listItems in the case where either of these fail.